src/NuProj.Targets/NuProj.targets

```diff
@@ -44,57 +42,17 @@
 
-  <!-- Set output and intermediate output properties. -->
-  <PropertyGroup>
-    <OutputPath Condition="'$(OutputPath)' != '' and !HasTrailingSlash('$(OutputPath)')">$(OutputPath)\</OutputPath>
-
-    <OutDirWasSpecified Condition=" '$(OutDir)'!='' and '$(OutDirWasSpecified)'=='' ">true</OutDirWasSpecified>
-    <OutDir Condition="'$(OutDir)' == ''">$(OutputPath)</OutDir>
-    <OutDir Condition="'$(OutDir)' != '' and !HasTrailingSlash('$(OutDir)')">$(OutDir)\</OutDir>
-
-    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">obj\</BaseIntermediateOutputPath>
-    <BaseIntermediateOutputPath Condition="!HasTrailingSlash('$(BaseIntermediateOutputPath)')">$(BaseIntermediateOutputPath)\</BaseIntermediateOutputPath>
-
-    <IntermediateOutputPathWasSpecified Condition=" '$(IntermediateOutputPath)'!='' and '$(IntermediateOutputPathWasSpecified)'=='' ">true</IntermediateOutputPathWasSpecified>
-    <IntermediateOutputPath Condition="$(IntermediateOutputPath) == ''">$(BaseIntermediateOutputPath)$(Configuration)\</IntermediateOutputPath>
-    <IntermediateOutputPath Condition="!HasTrailingSlash('$(IntermediateOutputPath)')">$(IntermediateOutputPath)\</IntermediateOutputPath>
-  </PropertyGroup>
-
-  <!-- For projects that want a per-project output directory, update OutDir and IntermediateOutputPath to include the project name. -->
-  <PropertyGroup>
-    <ProjectName Condition=" '$(ProjectName)' == '' ">$(MSBuildProjectName)</ProjectName>
-    <OutDir Condition="'$(OutDir)' != '' and '$(OutDirWasSpecified)' == 'true' and '$(GenerateProjectSpecificOutputFolder)' == 'true'">$(OutDir)$(ProjectName)\</OutDir>
-    <IntermediateOutputPath Condition="'$(IntermediateOutputPath)' != '' and '$(IntermediateOutputPathWasSpecified)' == 'true' and '$(GenerateProjectSpecificOutputFolder)' == 'true'">$(IntermediateOutputPath)$(ProjectName)\</IntermediateOutputPath>
-  </PropertyGroup>
-
-  <PropertyGroup>
-    <TargetDir Condition="'$(OutDir)' != ''">$([MSBuild]::Escape($([System.IO.Path]::GetFullPath(`$([System.IO.Path]::Combine(`$(MSBuildProjectDirectory)`, `$(OutDir)`))`))))</TargetDir>
-  </PropertyGroup>
-
-  <!--
-      NUSPEC PATH
-  -->
-
-  <PropertyGroup>
-    <NuSpecPath>$(IntermediateOutputPath)$(Id).nuspec</NuSpecPath>
-  </PropertyGroup>
-
-  <!--
-      OUTPUT PATHS
-
-      These properties aren't passed to NuGet.exe - they're implicit. However, we need to know the output paths
-      at several occasions (e.g. incremental build or clean up) so we want a central spot to capture those.
-
-      We set these properties in a Target so that other targets that determine what $(Version)
-      should be can run first.
-  -->
-
-  <Target Name="EstablishNuGetPaths" DependsOnTargets="$(VersionDependsOn)">
   <PropertyGroup>
-    <NuGetOutputPath>$(TargetDir)$(Id).$(Version).nupkg</NuGetOutputPath>
-    <NuGetSymbolsOutputPath>$(TargetDir)$(Id).$(Version).symbols.nupkg</NuGetSymbolsOutputPath>
+    <Language>NuGet</Language>
```

Comment:
So I figured out more about why this is a problem. The value of the `Language` property in MSBuild didn't previously matter to the two MSBuild extension I wrote that generates code via CodeDOM based on the `Language` proprety. It matters now, with your changes, because bringing in Microsoft.Common.targets includes bringing in the `PrepareResources` target, and the chain that actually executes it during a build. Prior to that point, that target wasn't defined nor executed, so some of my targets didn't either, which were only executed because `PrepareResources` was.  So in effect, NuProj subtly and automagically "worked" with these extensions by not invoking *part* of the extension. 
So to make this work for my extensions, I need to figure out how to tell whether an assembly will be compiled by a project. That way I can simulate the old effect that NuProj had of skipping that part of the build in my extension.
In all your studying of the .targets as part of this work, did you come across any such MSBuild property that might hint that no assembly is compiled?